dora/core/common/src/main/java/alluxio/membership/WorkerClusterSnapshot.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.membership;
+
+import alluxio.collections.IndexDefinition;
+import alluxio.collections.IndexedSet;
+import alluxio.wire.WorkerIdentity;
+import alluxio.wire.WorkerInfo;
+
+import com.google.common.collect.Iterators;
+
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Snapshot for a cluster view.
+ */
+public final class WorkerClusterSnapshot implements WorkerClusterView {
```

Comment:
i have the feeling that we don't really need this class? We can further discuss in a quick sync.

Reply:
done. Now `WorkerClusterSnapshot` replaces the `WorkerClusterView`.